### ISSUE TITLE
nonrecognition of inequality operators

### ISSUE THREAD

**Issue by amotzop** [microsoft/pyright#4636]
**Describe the bug**
When defining the method `__ge__`/`__le__` in a class, pyright recognises only the usage `instance >= other` and not `other <= instance` (and vice-versa for `__le__`)

**To Reproduce**
Define a class with a `__ge__` and use it when the instance of the class is in the RHS of the `>=` operator.

**Expected behavior**
It should be recognised as a call to the defined `__ge__` function.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class Cls:
    def __ge__(self, other):
        ...
1 <= Cls()
```
pyright gives and error on the last line (`Operator "<=" not supported for types "Literal[1]" and "A"`)

**VS Code extension or command-line**
Happens in both of them (with the newest versions, e.g., 1.1.293)

**Comment by erictraut** [comment]
Pyright's behavior is correct here. The opposite of the `>=` operator is `<`, not `<=`. If a class implements `__ge__`, it will support both `>=` and `<` but not `<=`.

**Comment by amotzop** [comment]
This seems a bit weird.
In the following code there is a mismatch between what pyright says and what is returned by running it:
```python
class Cls:
    def __ge__(self, other):
        return ">="

    def __gt__(self, other):
        return "<"

reveal_type(Cls() >= 1)
reveal_type(1 <= Cls())
```

pyright reveals the types as `Literal[">="]` and `Literal["<"]`, but running this code (replacing the `reveal_type` by `print`) prints ">=" and ">=".
(If `__ge__` wasn't defined then the second comparison wouldn't show as an error in pyright but raises an exception when ran)

**Comment by erictraut** [comment]
OK, I'll take a closer look.

**Comment by erictraut** [comment]
You are correct. My understanding was incorrect, and pyright was doing the wrong thing here.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.295, which I just published. It will also be included in a future release of pylance.
